### DISCUSSION TITLE
Cache error

### DISCUSSION THREAD

**Discussion by arnoutvanee** [theOehrly/Fast-F1#338] [Q&A]
Hi

I've been trying to get fastf1 to work for about a week now. I am new to working with Python so I have done some reading on how to get fastf1 to work. I use Pycharm for my projects. As far as I know I have installed all pips needed. However I keep running into the same problem.

When using the first simple code ...

`import fastf1`
`session = fastf1.get_session(2019, 'Monza', 'Q')`
`session.load(telemetry=False, laps=False, weather=False)`
`vettel = session.get_driver('VET')`
`print(f"Pronto {vettel['FirstName']}?")`

.... from the 'Getting Started' example, I **do** yield the result 'Pronto Sebastian?', however with the following message:

_api         WARNING_ 	

_NO CACHE! Api caching has not been enabled!_
	_It is highly recommended to enable this feature for much faster data loading!_
	_Use `fastf1.Cache.enable_cache('path/to/cache/')`_

_core           INFO 	Loading data for Italian Grand Prix - Qualifying [v2.3.1]_
_api            INFO 	Fetching driver list..._
_core        WARNING 	Failed to load data from Ergast API! (This is expected for recent sessions)_
_api            INFO 	Fetching race control messages..._
_core           INFO 	Finished loading data for 20 drivers: ['5', '18', '8', '20', '7', '4', '10', '11', '55', '26', '99', '23', '27', '44', '3', '16', '77', '33', '63', '88']_
_Pronto Sebastian?_

_Process finished with exit code 0_

When I include `fastf1.Cache.enable_cache('path/to/cache/')` as recommended I get **no** result but I do get the following error code:

_Traceback (most recent call last):_
  _File "C:\Users\arnou\PycharmProjects\pythonProject2\pip\main.py", line 2, in <module>_
   _fastf1.Cache.enable_cache('path/to/cache/')_
  _File "C:\Users\arnou\PycharmProjects\pythonProject2\pip\venv\lib\site-packages\fastf1\api.py", line 137, in enable_cache
    raise NotADirectoryError("Cache directory does not exist! Please check for typos or create it first.")_
_NotADirectoryError: Cache directory does not exist! Please check for typos or create it first._

_Process finished with exit code 1_

And I have been getting the same error code with whichever other code I have taken from the https://theoehrly.github.io/Fast-F1/ page.

So it seems I am doing something wrong relating to the cache...

I have been searching endlessly online how to fix this but I can't find a solution to this error code.

Can you please help me?

I am looking forward tremendously to start working with fastf1, lifelong F1 fan!

Many thanks in advance.

**Comment by theOehrly** [comment]
```
fastf1.Cache.enable_cache('path/to/cache/')
```
Here, `path/to/cache/` needs to be replaced with an actual valid file path to a cache directory and you need to create that directory first.

**Reply by arnoutvanee** [reply]
Thanks so much for your quick response!

I had been playing around with that before, but with no success.

Where does the folder need to be located? Or can it be located anywhere on de OS (C:) drive?

I have created a folder in the following location: C:\Users\arnou\PycharmProjects\pythonProject2\pip\cache

How should the code `fastf1.Cache.enable_cache('path/to/cache/')` be structured to get it to work?  

Thanks again.

**Reply by arnoutvanee** [reply]
I think I found it... I needed to change \ to / .... thats why it didn't work before. As only someone familiar with F1 will understand... 'I am stupid' ;)

Let's get going!

**Reply by MooshuBeef** [reply]
@arnoutvanee I just ran into the same error but the issue I had was I didn't format the path correctly and was missing a tilde (~) and needed to create the folder called "cache" manually within the `venv` folder. I ended up using: 

`fastf1.Cache.enable_cache('~/Documents/Formula1_python/venv')`
